python/rikai/spark/sql/codegen/base.py

```diff
@@ -74,7 +75,9 @@ def resolve(self, spec: "ModelSpec"):
             "type": "object",
             "properties": {
                 "pre": {"type": "string"},
+                "preCode": {"type": "string"},
```

Comment:
if we can use bytes directly we just can just use "pre" and have a list of acceptable types?

Reply:
To pass model spec from jvm to python, so we are writing the spec into a json , and let python to load it. 
That's where the `bytes` need to be encoded into string.

But it might be feasible to do base64 encode / decode right at the json serialization step. I can take a look